### DISCUSSION TITLE
Rekognition

### DISCUSSION THREAD

**Discussion by yuribudilov** [awslabs/aws-sdk-rust#274] [Q&A] [answered]
hello everyone

I am new to face recognition but I have worked with AWS and now learning Rust.

Can I use this AWS SDK to work with Rust and AWS Rekognition for face recognition in JPG photo files?
If not, what other options do I have from Rust and AWS ?

Use-case: I have >100,000+ JPG photos (and growing) and I have a list of ~10 individual faces to search for.
I want to search for specific faces in a pool of JPG photos and to report which of those individual faces exist in each one of >100,000 JPG photo files. If multiple faces exist in any one JPG, I want to report each face found in each JPG file.

This is not yet for production use, only for development and testing, I understand this SDK is not supported for production.
If there are some examples for Rekognition, I would appreciate it.
The current examples folder here on github does not cover Rekognition.

Many thanks for any guidance

**Comment by Velfi** [comment, marked as answer]
We're still working on creating more examples for the many services we support. In the mean time, check out the [detect_faces](https://awslabs.github.io/aws-sdk-rust/aws_sdk_rekognition/client/struct.Client.html#method.detect_faces) method in the Rekognition client. You can see the different inputs for that operation [here](https://awslabs.github.io/aws-sdk-rust/aws_sdk_rekognition/client/fluent_builders/struct.DetectFaces.html). You can learn about how to create a client by looking at current examples like [this one for S3](https://github.com/awslabs/aws-sdk-rust/blob/main/sdk/examples/s3/src/bin/s3-helloworld.rs) and then switching out the S3 parts for `aws_sdk_rekognition`. Good luck!

**Reply by yuribudilov** [reply]
thank you. 
It worked.

I have used examples from S3 and Rust AWS SDK source code to figure it out and I used compare_faces() from Rekognition to match a given face in JPG file against another set of JPG files to find a match.

Snippets of code:

```
    let rek_client = aws_sdk_rekognition::Client::new(&shared_config);
    let i = Image::builder().set_s3_object(Some(s3o)).build();
    let t = Image::builder().set_s3_object(Some(s3t)).build();

    let r = rek_client
        .compare_faces()
        .source_image(i)
        .target_image(t)
        .send()
        .await;


```

I am sure **I have not used Rust AWS SDK in the idiomatic/recommended way** so in future it would help me if AWS Examples included Rekognition to tell me how best to use it from Rust AWS SDK.

Thank you for your support, so far it is working for me and it did not take too long to get it working for this simple use case.
